### DISCUSSION TITLE
How to add Buy icon within the AR mode, and how to do some action when clicked.

### DISCUSSION THREAD

**Discussion by rksambari** [google/model-viewer#3576] [Q&A]
How to add Buy icon within the AR mode, and how to do some action when clicked.
### Description

I have an AR Model developed with model viewer, when I click on View in you space, I'm able to see the model in real world but I want to add an icon some where right/left of the camera view, and need to perform some action when clicked.

### Version
https://unpkg.com/focus-visible@5.0.2/dist/focus-visible.js
https://unpkg.com/@google/model-viewer/dist/model-viewer.min.js
- model-viewer: v5.0.2

### Browser Affected

- [x] Chrome, version: xx.x.xxxx.xx
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [ ] Safari

### OS

- [x] Android
- [x] iOS
- [ ] Linux
- [ ] MacOS
- [x] Windows

### AR

- [ ] WebXR
- [x] SceneViewer
- [ ] QuickLook

**Comment by elalish** [comment]
It depends on your `ar-mode`, which also depends on your device. For WebXR (our default on Android) this is easy; just add any HTML elements as children of your mv element: https://modelviewer.dev/examples/augmentedreality/index.html#webXR

For iOS, quick-look is your only option and you'll need to pass their API's query params in your `ios-src`. They don't have a lot of options, but you can create a button.

**Reply by rksambari** [reply]
When ar-mode is WebXR, adding html button inside mv element will give me the button in 3D view. But the button is not visible after I enter into AR. I want to that button real inside AR not just on the 3D view.

As you can see the images below, I want the button/buy icon should appear on the **ARView**.

**AR View**
![ARview](https://user-images.githubusercontent.com/99704624/177912792-10614821-d215-4a58-958c-593ae9f2656d.jpeg)

**3D View**
![3dview](https://user-images.githubusercontent.com/99704624/177912796-b912f6b3-62ae-43b0-aee0-e16c425f4132.jpeg)

**Reply by elalish** [reply]
It appears that you're using the scene-viewer `ar-mode`. This will work better in webxr mode. Try restricting your `ar-modes` to "webxr"; it may just be failing (you'll need HTTPS, for instance).

**Reply by rksambari** [reply]
@elalish Thanks for the response.

Can you please share if you have any example code/sample where I get ICON on the AR camera using webxr?
Thanks.

**Reply by elalish** [reply]
The first link I shared is that example. It has a whole HTML carousel, but you should be able to simplify that to an icon.

**Reply by rksambari** [reply]
Hi @elalish, we have 2 views in model viewer. 
   **ONE**, the 3d view which is the html page loads when you click on the url and this view has that AR icon(View In AR).
   **SECOND**, the AR view which comes up on clicking View in AR button in 3d view.

The carousel example you shared is the type **ONE**(3d View) but not the AR view. What I need is an ICON on the AR camera view(**SECOND**, the real-world camera view).

In my previous comment, I shared 2 images, one is taken from ARView(real-world camera view), second image is from 3d view. Your example is getting all items(carousel slider) on the 3d view but not in the AR view, thus can you please share an example that helps me creating an ICON on the AR View(**SECOND**).

Thanks.

**Reply by elalish** [reply]
Not true; if you try that example on Android and hit the AR button, you will see that same carousel in AR, via WebXR. You can style your CSS to change visibility and such between 3D and AR mode using this: https://modelviewer.dev/docs/#entrydocs-augmentedreality-css-arStatus.

**Reply by polishchuk37a** [reply]
> Not true; if you try that example on Android and hit the AR button, you will see that same carousel in AR, via WebXR. You can style your CSS to change visibility and such between 3D and AR mode using this: https://modelviewer.dev/docs/#entrydocs-augmentedreality-css-arStatus.

But how can i make it working for IOS devices?

**Reply by elalish** [reply]
You can't, sadly. Ask Apple to support WebXR? @grorg.
